codec-http/src/main/java/io/netty/handler/codec/spdy/SpdyHttpDecoder.java

```diff
@@ -357,27 +358,36 @@ private static FullHttpRequest createHttpRequest(int spdyVersion, SpdyHeadersFra
         headers.remove(PATH);
         headers.remove(VERSION);
 
-        FullHttpRequest req = new DefaultFullHttpRequest(httpVersion, method, url);
-
-        // Remove the scheme header
-        headers.remove(SCHEME);
+        boolean release = true;
+        ByteBuf buffer = alloc.buffer();
+        try {
+            FullHttpRequest req = new DefaultFullHttpRequest(httpVersion, method, url, buffer);
 
-        // Replace the SPDY host header with the HTTP host header
-        CharSequence host = headers.get(HOST);
-        headers.remove(HOST);
-        req.headers().set(HttpHeaderNames.HOST, host);
+            // Remove the scheme header
+            headers.remove(SCHEME);
 
-        for (Map.Entry<CharSequence, CharSequence> e: requestFrame.headers()) {
-            req.headers().add(e.getKey(), e.getValue());
-        }
+            // Replace the SPDY host header with the HTTP host header
+            CharSequence host = headers.get(HOST);
+            headers.remove(HOST);
+            req.headers().set(HttpHeaderNames.HOST, host);
 
-        // The Connection and Keep-Alive headers are no longer valid
-        HttpUtil.setKeepAlive(req, true);
+            for (Map.Entry<CharSequence, CharSequence> e : requestFrame.headers())
+            {
+                req.headers().add(e.getKey(), e.getValue());
+            }
 
-        // Transfer-Encoding header is not valid
-        req.headers().remove(HttpHeaderNames.TRANSFER_ENCODING);
+            // The Connection and Keep-Alive headers are no longer valid
+            HttpUtil.setKeepAlive(req, true);
 
-        return req;
+            // Transfer-Encoding header is not valid
+            req.headers().remove(HttpHeaderNames.TRANSFER_ENCODING);
+            release = false;
+            return req;
+        } finally {
+            if (release) {
+                buffer.release();
+            }
+        }
     }
 
     private static FullHttpResponse createHttpResponse(ChannelHandlerContext ctx, SpdyHeadersFrame responseFrame,
```

Comment:
consider making this signature consistent with the signature for `createHttpRequest`. Something like:

```java
private static FullHttpResponse createHttpResponse(SpdyHeadersFrame responseFrame,
                                                       boolean validateHeaders,
ByteBufAllocator alloc) {
..
}
```